packages/mdc-slider/index.ts

```diff
@@ -0,0 +1,186 @@
+      registerResizeHandler: (handler) => window.addEventListener('resize', handler),
+      deregisterResizeHandler: (handler) => window.removeEventListener('resize', handler),
+      notifyInput: () => this.emit(strings.INPUT_EVENT, this),
+      notifyChange: () => this.emit(strings.CHANGE_EVENT, this),
+      setThumbContainerStyleProperty: (propertyName, value) => {
+        this.thumbContainer_.style.setProperty(propertyName, value);
+      },
+      setTrackStyleProperty: (propertyName, value) => this.track_.style.setProperty(propertyName, value),
+      setMarkerValue: (value) => this.pinValueMarker_.innerText = value.toLocaleString(),
+      appendTrackMarkers: (numMarkers) => {
+        const frag = document.createDocumentFragment();
+        for (let i = 0; i < numMarkers; i++) {
+          const marker = document.createElement('div');
+          marker.classList.add('mdc-slider__track-marker');
+          frag.appendChild(marker);
+        }
+        this.trackMarkerContainer_.appendChild(frag);
+      },
+      removeTrackMarkers: () => {
+        while (this.trackMarkerContainer_.firstChild) {
+          this.trackMarkerContainer_.removeChild(this.trackMarkerContainer_.firstChild);
+        }
+      },
+      setLastTrackMarkersStyleProperty: (propertyName, value) => {
+        // We remove and append new nodes, thus, the last track marker must be dynamically found.
+        const lastTrackMarker = this.root_.querySelector<HTMLElement>(strings.LAST_TRACK_MARKER_SELECTOR)!;
+        lastTrackMarker.style.setProperty(propertyName, value);
+      },
+      isRTL: () => getComputedStyle(this.root_).direction === 'rtl',
+    });
+    // tslint:disable:object-literal-sort-keys
+  }
+
+  initialSyncWithDOM() {
+    const origValueNow = this.parseFloat_(this.root_.getAttribute(strings.ARIA_VALUENOW), this.value);
+    const min = this.parseFloat_(this.root_.getAttribute(strings.ARIA_VALUEMIN), this.min);
+    const max = this.parseFloat_(this.root_.getAttribute(strings.ARIA_VALUEMAX), this.max);
+
+    // min and max need to be set in the right order to avoid throwing an error
+    // when the new min is greater than the default max.
+    if (min >= this.max) {
+      this.max = max;
+      this.min = min;
+    } else {
+      this.min = min;
+      this.max = max;
+    }
+
+    this.step = this.parseFloat_(this.root_.getAttribute(strings.STEP_DATA_ATTR), this.step);
+    this.value = origValueNow;
+    this.disabled = (
+        this.root_.hasAttribute(strings.ARIA_DISABLED) &&
+        this.root_.getAttribute(strings.ARIA_DISABLED) !== 'false'
+    );
+    this.foundation_.setupTrackMarker();
+  }
+
+  layout() {
+    this.foundation_.layout();
+  }
+
+  stepUp(amount = (this.step || 1)) {
+    this.value += amount;
+  }
+
+  stepDown(amount = (this.step || 1)) {
+    this.value -= amount;
+  }
+
+  private parseFloat_(str: string | null, defaultValue: number) {
+    const num = parseFloat(str!); // tslint:disable-line:ban
```

Comment:
i removed this `tslint:disable-line` - i dont think you need it

Reply:
When I remove it, I get:

```
ERROR: packages/mdc-slider/index.ts:178:17 - Calls to 'parseFloat' are not allowed. tsstyle#type-coercion
```